flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -781,6 +787,32 @@ class TaskManager(
     }
   }
 
+  private def handleRequestTaskManagerLog
+  (sender: ActorRef, requestType: LogTypeRequest, jobManager: ActorRef):
+  Unit = {
+    if (blobService.isDefined && jobManager.path.address.host.isDefined) {
```

Comment:
If the `blobService` is defined the `jobManager` should have a defined address as well. I think it would be better to use pattern matching to extract the blobService from the `Option`. This is more scalaesque.